src/api.jl

```diff
@@ -235,9 +235,11 @@ function delete_engine(ctx::Context, engine::AbstractString; kw...)
 end
 
 function delete_model(ctx::Context, database::AbstractString, engine::AbstractString, model::AbstractString; kw...)
-    tx = Transaction(ctx.region, database, engine, "OPEN"; readonly = false)
-    actions = [_make_delete_models_action([model])]
-    return _post(ctx, PATH_TRANSACTION; query = query(tx), body = body(tx, actions...), kw...)
+    return exec(ctx, database, engine, """def delete:rel:catalog:model["$(model)"] =rel:catalog:model["$(model)"]"""; readonly=false, kw...)
```

Comment:
This is actually a security vulnerability, since the name `model` may include injection attacks.

It's tricky to get this right. It used to be that this was sufficient: `$(repr(model))` instead of `"$(model)", but now there will still be issues if the name includes `%` since this an interpolation character in Rel.

I _think_ the right answer is that the SDK needs to provide an `escape_string_for_rel(str)` function.

I filed an issue for this here: https://github.com/RelationalAI/rai-sdk-issues/issues/68

Can you maybe add that as part of this PR? Does the problem make sense?

Reply:
nathan can you elaborate on how this in itself is a security vuln? i believe the vuln is if an app developer takes input directly from some UI and inserts it here without any sanitization? i agree that from a correctness pov, we should escape, but not sure about the vuln. also, didnt we agree that query args was the better way to handle this?

Reply:
yeah, i think my comment predates the slack discussion where we agreed on query args as the better way to handle this. I believe i had just forgotten about that when i proposed the above.

> i believe the vuln is if an app developer takes input directly from some UI and inserts it here without any sanitization?

yes, that's the vulnerability I had in mind. You're right that it's probably on the developer to sanitize the inputs if they come from users, but it seems like something we could do for them to close the potential attack vector. That's all i meant.